chain-abci/src/storage/account/tree.rs

```diff
@@ -0,0 +1,420 @@
+//! # Value with associated properties (e.g. min/max bounds)
+//! adapted from https://github.com/ChosunOne/merkle_bit examples (Merkle-BIT)
+//! Copyright (c) 2019, Josiah Evans (licensed under the MIT License and the Apache License, Version 2.0)
+//! Modifications Copyright (c) 2019, Foris Limited (licensed under the Apache License, Version 2.0)
+//!
+//! TODO: WIP usage -- disallow dead_code when new TX types are added to work with accounts
+#![allow(dead_code)]
+use blake2::{Blake2s, Digest};
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use chain_core::common::H256;
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+use starling::constants::KEY_LEN;
+use starling::merkle_bit::{BinaryMerkleTreeResult, MerkleBIT};
+use starling::traits::Hasher;
+use starling::traits::{
+    Branch, Data, Database, Decode, Encode, Exception, Leaf, Node, NodeVariant,
+};
+use starling::tree::tree_data::TreeData;
+use std::collections::HashMap;
+use std::error::Error;
+use std::mem;
+
+#[derive(Clone)]
+pub struct Blake2sHasher(Blake2s);
+impl Hasher for Blake2sHasher {
+    type HashType = Self;
+
+    #[inline]
+    fn new(_size: usize) -> Self {
+        let hasher = Blake2s::new();
+        Self(hasher)
+    }
+
+    #[inline]
+    fn update(&mut self, data: &[u8]) {
+        self.0.input(data);
+    }
+
+    #[inline]
+    fn finalize(self) -> [u8; KEY_LEN] {
+        let result = self.0.result();
+        let mut finalized = [0; KEY_LEN];
+        finalized.copy_from_slice(result.as_ref());
+        finalized
+    }
+}
+
+pub type TreeHasher = Blake2sHasher;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub struct Count(u64);
+
+impl Encodable for Count {
+    fn rlp_append(&self, s: &mut RlpStream) {
```

Comment:
Why not just this?

```
fn rlp_append(&self, s: &mut RlpStream) {
    self.0.rlp_append(s);
}
```

Reply:
that doesn't work -- RLP doesn't dictate the byte order. But I think we can make a macro for these wrapper types

Reply:
RLP is BigEndian by default. Here's the implementation https://docs.rs/rlp/0.3.0/src/rlp/impls.rs.html#122-127